homeassistant/components/broadlink/device.py

```diff
@@ -1,57 +1,172 @@
 """Support for Broadlink devices."""
+import asyncio
 from functools import partial
 import logging
 
+import broadlink as blk
 from broadlink.exceptions import (
+    AuthenticationError,
     AuthorizationError,
     BroadlinkException,
     ConnectionClosedError,
     DeviceOfflineError,
 )
 
-from .const import DEFAULT_RETRY
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_TIMEOUT, CONF_TYPE
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN, DOMAINS_AND_TYPES
+from .updater import get_update_coordinator
 
 _LOGGER = logging.getLogger(__name__)
 
+DEFAULT_PORT = 80
+
 
 class BroadlinkDevice:
     """Manages a Broadlink device."""
 
-    def __init__(self, hass, api):
+    def __init__(self, hass, config):
         """Initialize the device."""
         self.hass = hass
+        self.config = config
+        self.api = None
+        self.coordinator = None
+        self.fw_version = None
+        self.authorized = None
+        self.reset_jobs = []
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self.config.title
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the device."""
+        return self.config.unique_id
+
+    @staticmethod
+    async def async_update(hass, entry):
+        """Update the device and related entities.
+
+        Triggered when the device is renamed on the frontend.
+        """
+        # Update the name in the registry.
+        device_registry = await dr.async_get_registry(hass)
+        device = device_registry.async_get_device({(DOMAIN, entry.unique_id)}, set())
+        device_registry.async_update_device(device.id, name=entry.title)
+
+        # Update the name in the API and related entities.
+        device = hass.data[DOMAIN].devices[entry.entry_id]
```

Comment:
Please distinguish the device registry `device` from the broadlink `device` by using different variable names.

We usually refer to registry items as entries, ie a device registry entry or device entry.

Reply:
Ok.